src/coprocessor/codec/mysql/decimal.rs

```diff
@@ -1349,6 +1349,16 @@ impl Decimal {
             return Res::Ok(self);
         }
 
+        let upper = (DIGITS_PER_WORD * word_buf_len * 2) as isize;
+        if shift > upper {
```

Comment:
It seems that in the mysql's implementation, there isn't such logic?

Reply:
yes, and `tidb` would also panic by overflow for cases like` shift` is `-9223372036854775808` or `9223372036854775807`. I think it's a better way to return overflow or truncated here when we can clearly get this information from `shift`. This can both make the logic easier and avoid overflow